[FEATURE] Support ESLint 10

ESLint 10 is scheduled for release on February 6th, but currently `html-eslint` can't be installed under `"eslint": "^10.0.0-rc.1"`:

```bash
npm error code ERESOLVE
npm error ERESOLVE unable to resolve dependency tree
npm error
npm error While resolving: undefined@undefined
npm error Found: eslint@10.0.0-rc.1
npm error node_modules/eslint
npm error   dev eslint@"^10.0.0-rc.1" from the root project
npm error
npm error Could not resolve dependency:
npm error peer eslint@"^8.0.0 || ^9.0.0" from @html-eslint/eslint-plugin@0.53.0
npm error node_modules/@html-eslint/eslint-plugin
npm error   dev @html-eslint/eslint-plugin@"^0.53.0" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
```

--- a/packages/eslint-plugin/lib/rules/utils/source-code.js
+++ b/packages/eslint-plugin/lib/rules/utils/source-code.js
@@ -2,7 +2,14 @@
 
 /** @param {Context<any[]>} context */
 function getSourceCode(context) {
-  return context.sourceCode || context.getSourceCode();
+  return (
+    context.sourceCode ||
+    /**
+     * @type {Context<any[]> & {
+     *   getSourceCode(): Context<any>["sourceCode"];
+     * }}
+     */ (context).getSourceCode()
+  );
 }
 
 module.exports = {

--- a/packages/eslint-plugin/package.json
+++ b/packages/eslint-plugin/package.json
@@ -46,7 +46,7 @@
     "@html-eslint/types": "^0.54.0"
   },
   "peerDependencies": {
-    "eslint": ">=8.0.0"
+    "eslint": ">=8.0.0 || ^10.0.0-0"
   },
   "devDependencies": {
     "@eslint/core": "^1.0.0",

